src/Parsers/Access/ParserGrantQuery.cpp

```diff
@@ -284,17 +283,36 @@ bool ParserGrantQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
 
     AccessRightsElements elements;
     std::shared_ptr<ASTRolesOrUsersSet> roles;
-    if (!parseElementsWithoutOptions(pos, expected, elements) && !parseRoles(pos, expected, is_revoke, attach_mode, roles))
-        return false;
 
-    if (cluster.empty())
+    bool current_grants = false;
+    if (!is_revoke && cluster.empty() && ParserKeyword{"CURRENT GRANTS"}.ignore(pos, expected))
+    {
+        current_grants = true;
+        if (ParserToken(TokenType::OpeningRoundBracket).ignore(pos, expected))
+        {
+            if (!parseElementsWithoutOptions(pos, expected, elements) && !parseRoles(pos, expected, is_revoke, attach_mode, roles))
+                return false;
+
+            if (!ParserToken(TokenType::ClosingRoundBracket).ignore(pos, expected))
+                return false;
+        }
+        else
+            elements.emplace_back(AccessType::ALL);
+    }
+    else
+    {
+        if (!parseElementsWithoutOptions(pos, expected, elements) && !parseRoles(pos, expected, is_revoke, attach_mode, roles))
+            return false;
+    }
+
+    if (cluster.empty() && !current_grants)
```

Comment:
This way `GRANT CURRENT GRANT OPTIONS ON CLUSTER` will show a quite unfriendly error message. Probably it's better to parse it and throw exceptions in the interpreter later.